### ISSUE TITLE
[Lua] Generated Lua C++ code causes warnings using VS2013

### ISSUE THREAD

**Issue by jpaulovich** [swig/swig#465]
After upgrading to SWIG 3.0.6 from SWIG 2.0.9 I noticed that the `SWIG_Lua_InstallConstants` method is now causing warnings when using Visual Studio 2013.  

Warnings:
warning C4244: 'argument' : conversion from 'lua_Number' to 'lua_Integer', possible loss of data
warning C4244: 'initializing' : conversion from 'long' to 'char', possible loss of data

The first warning refers to an improper cast from lua_Number to a method that expects an integer.  

As of version 3.0.6:

```
    case SWIG_LUA_INT:
      lua_pushstring(L,constants[i].name);
      lua_pushinteger(L,(lua_Number)constants[i].lvalue);
      lua_rawset(L,-3);
      break;
```

Fix should be:

```
    case SWIG_LUA_INT:
      lua_pushstring(L,constants[i].name);
      lua_pushinteger(L,(lua_Integer)constants[i].lvalue);
      lua_rawset(L,-3);
      break;
```

The second warning is simply missing the (char) cast.
As of version 3.0.6:

```
    case SWIG_LUA_CHAR:
      lua_pushstring(L,constants[i].name);
      {
        char c = (char)constants[i].lvalue;
        lua_pushlstring(L,&c,1);
      }
      lua_rawset(L,-3);
      break;
```

Fix should be:

```
    case SWIG_LUA_CHAR:
      lua_pushstring(L,constants[i].name);
      {
        char c = (char)constants[i].lvalue;
        lua_pushlstring(L,&c,1);
      }
      lua_rawset(L,-3);
      break;
```

Thanks!

**Comment by wsfulton** [comment]
@jpaulovich, we don't test Lua on Windows but if you are familiar with this environment we'd appreciate your help. If you know how to obtain Lua and compile C/C++ extensions with Visual C++, some info on how to do this in Doc\Manual\Windows.html would help. Also some Visual Studio project files in Examples\lua\class and Examples\lua\simple would be really helpful. Finally we could integrate testing into our Appveyor testing with warnings as errors to stop this happening again.

**Comment by ojwb** [comment]
Both were fixed by e66d8125aae043fc9a3b145116339ea5066ebf4c back in 2017.
